hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodieWriterUtils.scala

```diff
@@ -206,6 +206,18 @@ object HoodieWriterUtils {
     }
   }
 
+  /**
+   * In some contexts table config and the datasource write config are interchangeable.
+   * This is not usually the case so don't use this unless you are sure it is ok.
+   * */
+  def fixKeyGeneratorClassConfig(params: Map[String, String]): Map[String, String] = {
```

Comment:
how about make it private?
Also, can you rename to something like `inferKeyGeneratorClassConfig`?

Reply:
Can't make it private because it is used in other classes. I renamed it though

Reply:
Also rebased